### DISCUSSION TITLE
Error providing the ipa.

### DISCUSSION THREAD

**Discussion by Jroman997** [codemagic-ci-cd/codemagic-docs#1653] [Q&A]
I have the following configuration in codemagic.yml, This is a native ios app

workflows:
  native-ios:
    name: iOS
    instance_type: mac_mini_m1
    environment:
      ios_signing:
        distribution_type: development
        bundle_identifier: pe.com.xxxx.dev
      vars:
        BUNDLE_ID: pe.com.xxxx.dev
        XCODE_PROJECT: "Mobile/Mobile.xcodeproj"
        XCODE_WORKSPACE: "Mobile/Mobile.xcodeproj/project.xcworkspace" # PUT YOUR WORKSPACE NAME HERE
        XCODE_SCHEME: "MobileDev" # PUT THE NAME OF YOUR SCHEME HERE
        APP_ID: 1234567890
      xcode: latest
      cocoapods: default
    triggering:
      events:
        - push
        - pull_request
      branch_patterns:
        - pattern: "developer"
          include: true
          source: true
    scripts:
      - name: Set up code signing settings on Xcode project
        script: |
          xcode-project use-profiles
      - name: cat export_options.plist
        script: |
          cat /Users/builder/export_options.plist
      - name: Build ipa
        script: |
          xcode-project build-ipa --workspace "$XCODE_WORKSPACE" --scheme "$XCODE_SCHEME"
    artifacts:
      - build/ios/ipa/*.ipa
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM

When the command is executed xcode-project build-ipa --workspace "$XCODE_WORKSPACE" --scheme "$XCODE_SCHEME"

Using Xcode 14.3 (14E222b)
Check project.xcworkspace build settings
Execute "xcodebuild -workspace Mobile/Mobile.xcodeproj/project.xcworkspace -scheme MobileDev -showBuildSettings"

Archive project.xcworkspace
Execute "xcodebuild -workspace Mobile/Mobile.xcodeproj/project.xcworkspace -scheme MobileDev -archivePath build/ios/xcarchive/project_362iosmg.xcarchive archive COMPILER_INDEX_STORE_ENABLE=NO DEVELOPMENT_TEAM=XXXXX 'CODE_SIGN_IDENTITY=Apple Development' CODE_SIGN_STYLE=Manual"

Using Xcode 14.3 (14E222b)
Check project.xcworkspace build settings
Execute "xcodebuild -workspace Mobile/Mobile.xcodeproj/project.xcworkspace -scheme MobileDev -showBuildSettings"

Archive project.xcworkspace
Execute "xcodebuild -workspace Mobile/Mobile.xcodeproj/project.xcworkspace -scheme MobileDev -archivePath build/ios/xcarchive/project_362iosmg.xcarchive archive COMPILER_INDEX_STORE_ENABLE=NO DEVELOPMENT_TEAM=XXXXX 'CODE_SIGN_IDENTITY=Apple Development' CODE_SIGN_STYLE=Manual"

    Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'MobileDev' from project 'Mobile')
Failed to archive Mobile/Mobile.xcodeproj/project.xcworkspace

Build failed :|
Step 5 script `Build ipa` exited with status code 65

**Comment by icarusdust** [comment]
Hi @Jroman997, I answered your query in Intercom, let's continue there please.

**Reply by icarusdust** [reply]
The main issue here was the following error:

`error: Provisioning profile "XXXXXXXXX  Profile" doesn't include the currently selected device "builder Virtual Machine" (identifier 0000000000000000). (in target 'XXXXXXX' from project 'XXXXXXX')`

The solution was to add the following flag to the build command:

`xcode-project build-ipa --workspace "$XCODE_WORKSPACE"  --scheme "$XCODE_SCHEME"  --archive-flags "-destination 'generic/platform=iOS' `

As a side note, in order to generate more of Xcode logs, one can add `- /tmp/xcodebuild_logs/*.log` under the artefacts section in `codemagic.yaml`

**Reply by Jroman997** [reply]
thank @icarusdust
